typing/types.mli

```diff
@@ -35,15 +35,19 @@ type atomic =
 type mutability =
   | Immutable
   | Mutable of
-      { modal_upper_bound : Mode.Alloc.Comonadic.Const.t
-        (** The upper bound of the new field value upon mutation. *)
+      { mode : Mode.Value.Comonadic.lr
```

Comment:
I'm a little surprised this isn't a `Const`. Do we need to store variables here?

Reply:
oh, so am I. I think maybe this was bad merge conflict resolution with some changes that were (maybe?) undone in `main`. I'll double check...

Reply:
oh, I think perhaps you're looking at the diff of a merge commit from `main` - `main` has `Mode.Value.Comonadic.lr` here (I think due to `let mutable`, but I'm not really sure).